src/lib/modules/specialconfigs/index.js

```diff
@@ -83,22 +83,33 @@ class SpecialConfigs {
     const self = this;
 
     this.embark.registerActionForEvent("contracts:deploy:afterAll", (cb) => {
-      let afterDeployCmds = self.config.contractsConfig.afterDeploy || [];
-      async.mapLimit(afterDeployCmds, 1, (cmd, nextMapCb) => {
-        async.waterfall([
-          function replaceWithAddresses(next) {
-            self.replaceWithAddresses(cmd, next);
-          },
-          self.replaceWithENSAddress.bind(self)
-        ], nextMapCb);
-      }, (err, onDeployCode) => {
-        if (err) {
-          self.logger.trace(err);
-          return cb(new Error("error running afterDeploy"));
-        }
+      if (typeof self.config.contractsConfig.afterDeploy === 'function') {
+        this.getAfterDeployLifecycleHookDependencies().then(dependencies => {
+          (async () => {
+            await self.config.contractsConfig.afterDeploy(dependencies);
```

Comment:
Notice that we always call lifecycle hook within `async/await` context.

This also means that lifecycle hooks have to either (optionally) return a Promise, or being written in async/await themselves.

@alaibe I looked into supporting both, promise support as well as callback support, unfortunately, I couldn't make both APIs work, without the user worrying about how their lifecycle hooks are being called.